datafusion/physical-expr/src/expressions/cast.rs

```diff
@@ -572,7 +572,7 @@ mod tests {
                 Some(15000),
                 Some(25000),
                 Some(30000),
-                Some(11234),
```

Comment:
I believe this is caused by https://github.com/apache/arrow-rs/pull/3021 (cc @liukun4515 ) where now arrow will round 1.123_456_8 to the nearest integer rather than simply truncating it

Reply:
I think it was actually https://github.com/apache/arrow-rs/pull/3000

Reply:
> I believe this is caused by [apache/arrow-rs#3021](https://github.com/apache/arrow-rs/pull/3021) (cc @liukun4515 ) where now arrow will round 1.123_456_8 to the nearest integer rather than simply truncating it

this changes is reasonable for me

Reply:
I have filled a new issue to talk about the `rounding` when cast decimal to a decimal with smaller scale in 
https://github.com/apache/arrow-rs/pull/3139 cc @tustvold @alamb